### ISSUE TITLE
Bug: Tweet sent out twice [to discuss] [maybe a bug]

### ISSUE THREAD

**Issue by jeyemwey** [b0xCH/Traveli#18]
Hi,

for whatever reason, my journey this morning was posted on Twitter twice.

* First tweet: https://twitter.com/confuzd_/status/844054323944480768
* Second tweet: https://twitter.com/confuzd_/status/844054332454703105

Without investigating any logs, it does look like an error to me.

**Comment by b0xCH** [comment]
There are two possible scenarios on this.

1: Generating & posting a map for Twitter takes around 2 seconds. If you accidentally push the checkin button twice, you will be checked in twice. (and therefore there will be two tweets.) There is already a check against that, but it triggers only if the user is currently underway. This will be fixed before release.

2: The Twitter API ran into a timeout while sending the response that could have led the used Twitter SDK to retry submitting a tweet. But this is unlikely. 

I'll keep this issue updated.

**Comment by jeyemwey** [comment]
Had this issue again, was sure *not* to trigger 1.. 2. seems unlikely too, but might be the way..

1. https://twitter.com/confuzd_/status/844196329677111297
2. https://twitter.com/confuzd_/status/844196349134471168

**Comment by dodomedia** [comment]
Hmmm... 
![image](https://cloud.githubusercontent.com/assets/2552560/24166418/73258392-0e73-11e7-8488-2583a593b8ba.png)

**Comment by b0xCH** [comment]
The check-in seems to be triggered twice. I checked myself onto the same trainride without having any of these effects. On what platform is this happening? I will check if there were multiple check-in requests from your client side, as there is almost no possible way to have such effects in the source code of this function.

**Comment by jeyemwey** [comment]
I am using the software with Chrome 56 on Android 7.1.

I found out that it might have actually been `1.` (sorry). Hence, we're having an Usability issue here.. In my view, the whole `button onClick listener` should be deactivated during the 2s-call to unpopulate this. Maybe, the method that darken the button can be extended for this.

**Comment by b0xCH** [comment]
Just verified this on the server-side. There are 2 calls to the checkin-function and therefore two checkins. I'll fix this on the client side to prevent this from happening again! :)

**Comment by b0xCH** [comment]
The submit-button to check in your trainride will be disabled after one click. This prevents multiple checkins.

**Comment by jeyemwey** [comment]
.[@saxoman](https://twitter.com/saxoman) also posted bis journey twice:

1. https://twitter.com/saxo_man/status/846717793328250880
2. https://twitter.com/saxo_man/status/846717803285659649

Maybe, this issue is not gone yet.

**Comment by b0xCH** [comment]
Definitely fixed it now on the server side. Its not possible to check in twice without waiting for one minute.

**Comment by jeyemwey** [comment]
I can't really tell if this is the latest app version, but this `alert()` is rather confusing than helping the user. Asking for "Double check-in?" when only having "Ok" to confirm is not really helpful.

A better solution might be "We noticed that you have already checked in this train. We stopped your request and did not send another tweet." as it states exactly what the system does.

![screenshot_20170406-144104](https://cloud.githubusercontent.com/assets/2796271/24754597/71865334-1ad7-11e7-9e34-23bdaee12ee0.png)
